### DISCUSSION TITLE
RRDstats SmartHome – fehlerhafte Anzeige Button Änderung Solltemperatur

### DISCUSSION THREAD

**Discussion by leo-022** [Freetz-NG/freetz-ng#371] [Unerfüllte Wünsche]
Bedingt durch die beiden von AVM vorgegebenen Werte 253 (aus) und 254 (ein) erfolgt am jeweiligen Button auf der Seite Status -> SmartHome eine falsche Anzeige der Temperatur:
![Button](https://user-images.githubusercontent.com/87432648/129862426-d537c33b-d8ce-4f4f-a505-a32b04e21769.png)
Statt 126,5°C müsste „aus“ und statt 127 „ein“ angezeigt werden. Dieses Verhalten tritt z. B. auf, wenn im AVM-Webinterface ein Heizung-aus-Zeitraum festgelegt wurde und dieser gerade aktiv ist.

Die Auswertung erfolgt in /usr/lib/cgi-bin/rrdstats/main.cgi ab Zeile 1390. Da der Wert durch 2 geteilt wird, erscheinen 126,5 und 127.
```
if [ -n "$tsoll" ]; then
   soll="$(echo $(( ${tsoll}0 / 2)) | sed 's/.$/,&/g')"
   temps=" (${grad%,0}->${soll%,0})"
```
Ebenso müsste auch im Eingabefenster die Eingabe der Werte 253 und 254 möglich sein (Multiplikation mit 2 nicht für diese beiden Zahlen).
```
butact="onclick=\"val = prompt('Solltemperatur (8-28):');if (val) window.location=('$SCRIPT_NAME?pdev=$pdev$refresh&ain=$ain&state='+parseFloat(val.replace(',','.'))*2)\""
```

**Comment by fda77** [comment]
Ich hab wie gesagt on/off nie benutzt. Müsste halt jemand das bestehende erweitern. Ich schiebs mal auf die Wishlist

**Comment by leo-022** [comment]
> Müsste halt jemand das bestehende erweitern.

Gut, da versuche ich mich mal daran. Es werden jetzt „AUS“ und „EIN“ angezeigt. Im Fenster zur Einstellung der Solltemperatur kann man zusätzlich zu den Gradzahlen noch 253 oder 254 (für AUS oder EIN) eingeben.

Einen Fehler im übrigen Quelltext habe ich noch beseitigt: Bei der Anzeige der Scheinleistung steht die nicht definierte Variable „NSBP“, sicher ein Schreibfehler.

siehe d20d447

**Comment by fda77** [comment]
Sieht doch gut aus. Mach doch gleich eine pull-request, das macht die Sache einfacher. Am besten noch in die 1. Zeile schrieben um welches Package es sich überhaupt handelt.
Ich hätte den NSBP-Typo korrigiert und nicht entfernt, passt die Ausrichtung mit den anderen Zeilen (ich weiss nicht so genau wo das angezeigt wird)

**Comment by leo-022** [comment]
Ich hatte NSBP erst korrigiert, da hatte es aber "Scheinleistung" um eine Position nach rechts verschoben. Ich denke, mit Entfernen sieht es richtig aus:
![scheinleistung](https://user-images.githubusercontent.com/87432648/130677460-041f95d9-13d9-4d64-be5a-e739247f7438.png)
(Anzeige unter Status -> SmartHome -> Wirkleistung oder Scheinleistung)

**Comment by fda77** [comment]
Komische Sache ^^ Vielleicht hatte ich das 3. Leerzeichn in Klammern gesetzt weil das 2. falsch geschriebene nichts bewirkt hat
